aidial_adapter_bedrock/utils/adapter_deployments.py

```diff
@@ -0,0 +1,144 @@
+            reference_deployment_id=self.reference_deployment_id,
+        )
+
+    def clone(self, reference_deployment_id: _T) -> "AdapterDeployment[_T]":
+        return AdapterDeployment(
+            adapter_deployment_id=self.adapter_deployment_id,
+            upstream_deployment_id=self.upstream_deployment_id,
+            reference_deployment_id=reference_deployment_id,
+        )
+
+
+AdapterChatCompletionDeployment = AdapterDeployment[ChatCompletionDeployment]
+AdapterEmbeddingsDeployment = AdapterDeployment[EmbeddingsDeployment]
+
+
+class AdapterDeployments(BaseModel):
+    chat_completions: Dict[str, AdapterChatCompletionDeployment]
+    embeddings: Dict[str, AdapterEmbeddingsDeployment]
+
+    @classmethod
+    def create(cls, *, compat_mapping: Dict[str, str]) -> "AdapterDeployments":
+
+        chat_completions = {e.value for e in ChatCompletionDeployment}
+        embeddings = {e.value for e in EmbeddingsDeployment}
+
+        for deployment_id, supported_id in compat_mapping.items():
+            if deployment_id in chat_completions or deployment_id in embeddings:
+                log.warning(
+                    f"{deployment_id!r} is one of the Bedrock deployments supported by the adapter already. "
+                    f"Remove {deployment_id!r} from the compatibility mapping to avoid the warning."
+                )
+
+                if (
+                    deployment_id in chat_completions
+                    and supported_id in embeddings
+                ):
+                    raise ValueError(
+                        f"The chat completion deployment {deployment_id!r} is mapped onto the embeddings deployment {supported_id!r}"
+                    )
+
+                if (
+                    deployment_id in embeddings
+                    and supported_id in chat_completions
+                ):
+                    raise ValueError(
+                        f"The embeddings deployment {deployment_id!r} is mapped onto the chat completion deployment {supported_id!r}"
+                    )
+
+        chat_completions = _create_deployments(
+            compat_mapping,
+            ChatCompletionDeployment,
+            redirects=CHAT_COMPLETION_REDIRECTS,
+        )
+        embeddings = _create_deployments(compat_mapping, EmbeddingsDeployment)
+
+        if compat_mapping:
+            raise ValueError(
+                f"None of the values in the following compatibility mapping corresponds to a Bedrock deployment supported by the adapter: {json.dumps(compat_mapping)}. "
+                f"Remap the deployments to the supported Bedrock deployments to fix the error."
+            )
+
+        ret = cls(chat_completions=chat_completions, embeddings=embeddings)
+
+        log.debug(f"Adapter deployments: {ret.json()}")
+
+        return ret
+
+
+def _create_deployments(
+    compat_mapping: Dict[str, str],
+    upstream_deployments: Iterable[_D],
+    *,
+    redirects: Dict[_D, _D] = {},
```

Comment:
Let's avoid usage of mutable objects as default for function arguments


```python
def _test(redirects: Dict = {}):
    print(redirects)
    # ... somewhere in code we modify dict
    redirects['test'] = 'value'
    # ...

_test()
# {}
_test()
# {"test": "value"}
```

Currently thay's not the case, since we only read this variable

But I would still prefer
```python
def _create_deployments(
    ...
    redirects: Dict[_D, _D] | None = None,
):
    if redirects is None:
        redirects = {}
```